pkg/services/teamgroupsync/teamgroupsync.go

```diff
@@ -0,0 +1,24 @@
+package teamgroupsync
+
+import (
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/registry"
+)
+
+type TeamSync interface {
+	SyncTeams(user *models.User, externalUser *models.ExternalUserInfo) error
+}
+
+type TeamSyncService struct{}
```

Comment:
```suggestion
type teamSyncStub struct{}
```

Personal preference, I don't think we need this to be public.